plugin/src/main/java/io/snyk/eclipse/plugin/views/SnykView.java

```diff
@@ -46,359 +46,359 @@
+		IActionBars bars = getViewSite().getActionBars();
+		fillLocalToolBar(bars.getToolBarManager());
+	}
+
+	private void fillContextMenu(IMenuManager manager) {
+		ITreeSelection selection = viewer.getStructuredSelection();
+		DisplayModel selected = (DisplayModel) selection.getFirstElement();
+		monitorActions = new ArrayList<>();
+		if (selected != null && selected.projectName != null) {
+			manager.add(rerunProjectAction(selected.projectName));
+			manager.add(monitorAction(selected.projectName));
+		}
+		if (selected != null && selected.id != null) {
+			manager.add(ignoreAction(selected.id, selected.iProject));
+		}
+		manager.add(scanWorkspace);
+		manager.add(new Separator());
+		manager.add(openPrefPage);
+
+	}
+
+	private void fillLocalToolBar(IToolBarManager manager) {
+		manager.add(scanWorkspace);
+		manager.add(abortScanning);
+	}
+
+	private void makeActions() {
+		openPrefPage = new Action() {
+
+			@Override
+			public void run() {
+				PreferenceDialog pref = PreferencesUtil.createPreferenceDialogOn(getShell(),
+						"io.snyk.eclipse.plugin.properties.preferencespage", null, null);
+				if (pref != null)
+					pref.open();
+			}
+		};
+		openPrefPage.setText("Preferences");
+
+		scanWorkspace = new Action() {
+			@Override
+			public void run() {
+				if (alreadyRunning)
+					return;
+				scanWorkspace.setEnabled(false);
+				abortScanning.setEnabled(true);
+				showMessage(RUNNING);
+
+				CompletableFuture.runAsync(() -> {
+					SnykExtendedLanguageClient.getInstance().triggerScan(null);
+					alreadyRunning = true;
+					List<DisplayModel> scanResult = DataProvider.INSTANCE.scanWorkspace();
+					rootModel.children.clear();
+					rootModel.children.addAll(scanResult);
+					viewer.getTree().getDisplay().asyncExec(() -> viewer.refresh());
+					scanWorkspace.setEnabled(true);
+					alreadyRunning = false;
+				});
+
+			}
+		};
+
+		scanWorkspace.setText("Snyk Test");
+		scanWorkspace.setToolTipText("Snyk Test");
+		scanWorkspace.setImageDescriptor(
+				Activator.getImageDescriptor("platform:/plugin/org.eclipse.ui.browser/icons/clcl16/nav_go.png"));
+
+		enableScanBasedOnConfig();
+
+		abortScanning = new Action() {
+			@Override
+			public void run() {
+				showMessage(ABORTING);
+				DataProvider.abort.set(true);
+				abortScanning.setEnabled(false);
+			}
+		};
+		abortScanning.setImageDescriptor(
+				Activator.getImageDescriptor("platform:/plugin/org.eclipse.ui.browser/icons/clcl16/nav_stop.png"));
+		abortScanning.setEnabled(false);
+	}
+
+	private static Shell getShell() {
+		var activeWorkbenchWindow = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
+		if (activeWorkbenchWindow == null)
+			return SHELL;
+		return activeWorkbenchWindow.getShell();
+	}
+
+	public void enableScanBasedOnConfig() {
+		try {
+			boolean cliFound = getCliFile().exists();
+			boolean tokenFound = !Preferences.getInstance().getAuthToken().isBlank()
+					|| System.getenv().containsKey("SNYK_TOKEN");
```

Comment:
The SNYK_TOKEN check is old and a bug :)

Reply:
Should we remove tokenFound, or only the System.getenv().containsKey check?